File: yeaji/lemonade-change.go
func lemonadeChange(bills []int) bool {
    changes := map[int]int{5:0, 10:0}
    for _, bill := range bills {
        if bill == 10 {
            if changes[5] >= 1 {
                changes[5]--
            } else{
                return false
            }
            changes[10]++
        } else if bill == 5 {
            changes[5]++
        } else { //20
            if changes[10] >0 && changes[5] > 0 {
                changes[10]--
                changes[5]--
            } else if changes[5] >= 3{
                changes[5] = changes[5] - 3
            } else {
                return false
            }
        }
    }
    return true
}

File: yeaji/lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int five = 0;
        int ten = 0;
        for (int temp : bills) {
            if (temp == 5){
                five ++;
                continue;
            } else if (temp == 10) {
                ten ++;
                five --;
            } else if (temp == 20) {
                if (ten >= 1){
                    ten--;
                    five--;
                } else {
                    five -= 3;
                }
            }
            
            if (five < 0 || ten < 0) {
                return false;
            }
        }
        return true;
    }
}
